setup/helpers/abs_helper.rb

```diff
@@ -654,24 +660,143 @@ def valid_abs_resource_hosts?(abs_resource_hosts)
+  #
+  # @raise [RuntimeError] If parsing is not successful
+  #
+  # @return [Array] The hosts array
+  #
+  # @example
+  #   hosts = parse_abs_resource_hosts(abs_resource_hosts)
+  #
+  def parse_abs_resource_hosts(abs_resource_hosts)
+    begin
+      hosts = JSON.parse(abs_resource_hosts)
+
+    # TODO: JSON::ParserError?
+    rescue StandardError => e
+      # TODO: raise? (current expected behavior is puts)
+      puts "#{e.inspect} encountered parsing the hosts array: #{abs_resource_hosts}"
+      hosts = nil
+    end
+    hosts
+  end
+
+  # Provisions a host with a tag for the specified role
+  # with the optionally specified size and volume_size
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] role The value to specify for the role tag
+  # @param [String] size The EC2 instance size to use
+  # @param [String] volume_size The volume size to use
+  #
+  # @raise [RuntimeError] If provisioning is not successful
+  #
+  # @return [String] The hostname of the provisioned host
+  #
+  # @example
+  #   hostname = provision_host_for_role(role)
+  #
+  #   or
+  #
+  #   hostname = provision_host_for_role(role, "c5.xlarge", "40")
+  #
+  # TODO: spec test(s)
+  def provision_host_for_role(role, size = AWS_DEFAULT_SIZE, volume_size = AWS_VOLUME_SIZE)
+    host_to_provision = get_host_to_provision(role, size, volume_size)
+    abs_resource_hosts = get_abs_resource_hosts(host_to_provision)
+    raise "Unable to provision host via ABS" unless abs_resource_hosts
+
+    hosts = parse_abs_resource_hosts(abs_resource_hosts)
+    hostname = hosts[0]["hostname"]
+
+    puts "Successfully provisioned host - role: #{role}, hostname: #{hostname}"
+    puts
+
+    hostname
+  end
+
+  # Provisions a set of hosts for the specified roles
+  # with the optionally specified prefix, size, and volume_size
+  #
+  # @author Bill Claytor
+  #
+  # @param [Array] roles The roles for which hosts should be provisioned
+  # @param [String] abs_id The value to specify for id tag
+  # @param [String] size The EC2 instance size to use
+  # @param [String] volume_size The volume size to use
+  #
+  # @return [Array<Hash>] The provisioned hosts
+  #
+  # @example
+  #   roles = ["master","puppet_db"]
+  #   hosts = provision_hosts_for_roles(roles)
+  #
+  #   or
+  #
+  #   hosts = provision_host_for_role(roles, "slv-test", "c5.xlarge", "40")
+  #
+  # TODO: spec test(s)
+  def provision_hosts_for_roles(roles, abs_id = nil, size = AWS_DEFAULT_SIZE, volume_size = AWS_VOLUME_SIZE)
+    hosts = []
+
+    puts "Provisioning hosts for the following pe_xl roles:"
+    puts roles
+    puts
+
+    # this will be evaluated in abs_initialize
+    ENV["ABS_ID"] = abs_id unless abs_id.nil?
+    num_threads = roles.length > MAX_THREADS ? roles.length : MAX_THREADS
+
+    # uses in_threads to allow variable modification
+    Parallel.map(roles.each, in_threads: num_threads) do |role|
+      # role = "#{abs_id}_#{role}" unless abs_id.nil?
```

Comment:
Should this be deleted?

Reply:
Yes; that was left over after an update to using a separate tag for the id.